### PR TITLE
Grouped menu feature @abp/ng.core

### LINKED ISSUE #15884: Angular UI to define groups for menu items
Angular UI of https://github.com/abpframework/abp/issues/15883

### LINKED ISSUE #15935: Feature #15884 (Grouped menu)
related #15884

- [x] Create optional group property type: string
- [x] Create **groupedTree** getter method to **RoutesService**
  -  Return grouped menu list if there is any group in tree
- [x] Add example to `Modifying-the-menu` docs 
- [x] Create test to routes.service.spec.ts
- [x] Add key for `AbpUi::OthersGroup` to [Volo.Abp.Ui](https://github.com/abpframework/abp/tree/dev/framework/src/Volo.Abp.UI/Localization/Resources/AbpUi) layer

### PR DESCRIPTION
**Pull request by masum-ulu**
Resolves #15884, resolves #15935

- [x]  I fully tested it as developer
- [x]  I documented it (or no need to document or I will create a separate documentation issue)

## Explanation
- **tree-utils.ts** updated 
  - `RouteGroup` type & `createGroupMap` function created
- [OTHERS_GROUP](https://github.com/abpframework/abp/blob/bfb3b0ccd9dbf674f4e79cadbb2cd9acacc447c3/npm/ng-packs/packages/core/src/lib/tokens/others-group.token.ts) Injection token created
- **RoutesService** updated
   - `othersGroup` field, `createGroupedTree` method added to **AbstractTreeService**
   - groupedTree and  groupedTree$ getter method added to `RoutesService`
- `routes.service.spec.ts` updated
  - Added new test for grouped menu items #groupedVisible
- **CoreModule** updated
   - added othersGroup property to options model
- **Modifying-the-Menu** document updated

# Example
- When we add a group to any route item, output should be looks like below
  - Create new group for home route item
  - Define group with `AbpIdentity::IdentityManagementGroup` key to identity route items
  - Define new key for `Others grup` with `OTHERS_GROUP` injection token or using with 
   `CoreModule.forRoot({ othersGroup: 'MyOthersGroup')` method
 
image

### CODE REVIEW

File: npm/ng-packs/packages/core/src/lib/core.module.ts [hunk 1]
```diff
@@ -176,6 +177,10 @@ export class CoreModule {
           provide: QUEUE_MANAGER,
           useClass: DefaultQueueManager,
         },
+        {
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/core/src/lib/core.module.ts:183]: I think the group might be optional. Depend on the behavior.  If I didn't add any group, menu behavior shouldn't change. isn't it?

File: npm/ng-packs/packages/core/src/lib/models/common.ts [hunk 2]
```diff
@@ -66,10 +68,21 @@ export namespace ABP {
...
+  export interface Group<TKey = string> {
+    key: TKey;
+    text: string;
+  }
+
   export interface Route extends Nav {
     path?: string;
     layout?: eLayoutType;
     iconClass?: string;
+    group?: Group<any>;
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/core/src/lib/models/common.ts]: Why we set any as group key type ? there is already a default value.

File: docs/en/UI/Angular/Modifying-the-Menu.md [hunk 3]
```diff
@@ -88,14 +88,65 @@ function configureRoutes(routes: RoutesService) {
...
+To get route items as grouped we can use `groupedVisible` (or Observable one `groupedVisible$`)  getter methods
+ - It will return `RouteGroup[]` if there is any group in route tree otherwise it'll return `undefined`
+
+```js
+import { ABP, RoutesService, RouteGroup } from "@abp/ng.core";
+import { Component } from "@angular/core";
+
+@Component(/* component metadata */)
+export class AppComponent {
+  visible: RouteGroup<ABP.Route>[] | undefined = this.routes.groupedVisible;
```
Comment (mahmut-gundogdu) [inline docs/en/UI/Angular/Modifying-the-Menu.md:126]: I didn't see code where the using visible fn or observe

**Review by mahmut-gundogdu — CHANGES_REQUESTED** [review]

File: npm/ng-packs/packages/core/src/lib/models/common.ts (hunk 2 above)
Comment (masum-ulu) [reply npm/ng-packs/packages/core/src/lib/models/common.ts]: Thi is old version of code, this is the [new version](https://github.com/abpframework/abp/blob/e21e8686fce1a05f77b135d2e0bd95c96faac096/npm/ng-packs/packages/core/src/lib/models/common.ts#L74)

File: npm/ng-packs/packages/core/src/lib/core.module.ts (hunk 1 above)
Comment (masum-ulu) [reply npm/ng-packs/packages/core/src/lib/core.module.ts:183]: It won't effect to default menu behavior, only using in groupedTree & groupedTree$ getter methods. This methods will return undefined when there is no any group in tree, [control line](https://github.com/abpframework/abp/blob/e21e8686fce1a05f77b135d2e0bd95c96faac096/npm/ng-packs/packages/core/src/lib/utils/tree-utils.ts#L81)

**Review by mahmut-gundogdu — APPROVED** [review]

**Review by braim23 — APPROVED** [review]

**Pull request merged**